enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -133,8 +134,30 @@ protected void addInitContainer(PodTemplateSpecBuilder builder, JSONObject initC
         String initContainerAnnotation = builder.buildMetadata().getAnnotations().get(INIT_CONTAINER_ANNOTATION);
         JSONArray initContainers = Strings.isNullOrBlank(initContainerAnnotation) ? new JSONArray()
                 : new JSONArray(initContainerAnnotation);
-        initContainers.put(initContainer);
 
+        // lets avoid duplicates being added
+        boolean duplicate = false;
+        int length = initContainers.length();
+        for (int i = 0; i < length; i++) {
+            JSONObject obj = initContainers.getJSONObject(i);
+            if (duplicateInitcontainer(obj, initContainer)) {
+                duplicate = true;
+            }
+
+        }
+        if (!duplicate) {
+            initContainers.put(initContainer);
+        }
         builder.editMetadata().addToAnnotations(INIT_CONTAINER_ANNOTATION, initContainers.toString()).endMetadata();
     }
+
+    private boolean duplicateInitcontainer(JSONObject a, JSONObject b) {
+        Object name1 = a.get("name");
+        Object name2 = b.get("name");
+        if (Objects.equal(name1, name2)) {
+            log.warn("Found duplicate init containers with names " + name1 + " so ignoring " + b);
```

Comment:
well if you're not sure how that can happen - just review the fabric8-platform git commit logs ;) f-m-p released a new version; then fabric8-platform's YAML became unusable.

Though I'm hoping we can setup CI tests to prevent this kinda thing ever happening again really - it happens very frequently with changes in kubernetes model, client & fmp. Now we actually have a working CD cluster finally; we can hopefully get that in place soon...